src/Infrastructure/DistGrid/src/ESMCI_DistGrid.C

```diff
@@ -5099,8 +5145,9 @@ template<typename T> int DistGrid::tGetSequenceIndexLocalDe(
       tArbSeqIndexListPCollPLocalDe[0][localDe][linExclusiveIndex]);
   }else{
     // determine the sequentialized index by construction of default tile rule
-    const int *localDeToDeMap = delayout->getLocalDeToDeMap();
-    int tile = tileListPDe[localDeToDeMap[localDe]];  // tiles are basis 1 !!!!
+    int tile = getTilePLocalDe(localDe, &localrc);
+    if (ESMC_LogDefault.MsgFoundError(localrc, ESMCI_ERR_PASSTHRU,
+      ESMC_CONTEXT, &rc)) return rc;
```

Comment:
I would especially welcome a close review of my new MsgFoundError error handling calls, here and elsewhere. I have tried to replicate similar error handling elsewhere, with particular attention given to the use of localrc vs. rc (which appears to be a bit inconsistent between different parts of the code). But it's possible that I'm misusing this error handling in some places.

Also note lack of error handling in some situations like in `GridIter::setDEBnds` (in ESMCI_Grid.C), which didn't seem to be set up to accommodate error handling.